src/observability/grafana/grafana_launcher.star

```diff
@@ -0,0 +1,144 @@
+
+TEMPLATES_FILEPATH = "./templates"
+
+DATASOURCE_UID = "grafanacloud-prom"
+DATASOURCE_CONFIG_TEMPLATE_FILEPATH = TEMPLATES_FILEPATH + "/datasource.yml.tmpl"
+DATASOURCE_CONFIG_REL_FILEPATH = "datasources/datasource.yml"
+
+CONFIG_DIRPATH_ON_SERVICE = "/config"
+
+USED_PORTS = {
+    HTTP_PORT_ID: ethereum_package_shared_utils.new_port_spec(
+        HTTP_PORT_NUMBER_UINT16,
+        ethereum_package_shared_utils.TCP_PROTOCOL,
+        ethereum_package_shared_utils.HTTP_APPLICATION_PROTOCOL,
+    )
+}
+
+
+def launch_grafana(
+    plan,
+    prometheus_private_url,
+    global_node_selectors,
+    grafana_params,
+):
+    datasource_config_template = read_file(DATASOURCE_CONFIG_TEMPLATE_FILEPATH)
+
+    grafana_config_artifact_name = upload_grafana_config(
+        plan,
+        datasource_config_template,
+        prometheus_private_url,
+    )
+
+    config = get_config(
+        grafana_config_artifact_name,
+        global_node_selectors,
+        grafana_params,
+    )
+
+    service = plan.add_service(SERVICE_NAME, config)
+
+    service_url = "http://{0}:{1}".format(
+        service.ip_address, service.ports[HTTP_PORT_ID].number
+    )
+
+    provision_dashboards(plan, service_url, grafana_params.dashboard_sources)
+
+    return service_url
+
+
+def upload_grafana_config(
+    plan,
+    datasource_config_template,
+    prometheus_private_url,
+):
+    datasource_data = new_datasource_config_template_data(prometheus_private_url)
+    datasource_template_and_data = ethereum_package_shared_utils.new_template_and_data(
+        datasource_config_template, datasource_data
+    )
+
+    template_and_data_by_rel_dest_filepath = {
+        DATASOURCE_CONFIG_REL_FILEPATH: datasource_template_and_data,
+    }
+
+    grafana_config_artifact_name = plan.render_templates(
+        template_and_data_by_rel_dest_filepath, name="grafana-config"
+    )
+
+    return grafana_config_artifact_name
+
+
+def new_datasource_config_template_data(prometheus_url):
+    return {"PrometheusUID": DATASOURCE_UID, "PrometheusURL": prometheus_url}
+
+
+def get_config(
+    grafana_config_artifact_name,
+    node_selectors,
+    grafana_params,
+):
+    return ServiceConfig(
+        image=grafana_params.image,
+        ports=USED_PORTS,
+        env_vars={
+            "GF_PATHS_PROVISIONING": CONFIG_DIRPATH_ON_SERVICE,
+            "GF_AUTH_ANONYMOUS_ENABLED": "true",
+            "GF_AUTH_ANONYMOUS_ORG_ROLE": "Admin",
+            "GF_AUTH_ANONYMOUS_ORG_NAME": "Main Org.",
+            # "GF_DASHBOARDS_DEFAULT_HOME_DASHBOARD_PATH": "/dashboards/default.json",
+        },
+        files={
+            CONFIG_DIRPATH_ON_SERVICE: grafana_config_artifact_name,
+        },
+        min_cpu=grafana_params.min_cpu,
+        max_cpu=grafana_params.max_cpu,
+        min_memory=grafana_params.min_mem,
+        max_memory=grafana_params.max_mem,
+        node_selectors=node_selectors,
+    )
+
+
+def provision_dashboards(plan, service_url, dashboard_sources):
```

Comment:
Since this relies on [grizzly](https://github.com/grafana/grizzly) existing, do we need to call that out in the README?
And/Or if there's a package manager involved, can we just add it as an explicit dependency?

Reply:
it uses the grizzly docker image, so no local dependency required